src/main/java/de/tum/in/test/api/ast/model/JavaFile.java

```diff
@@ -26,11 +28,22 @@ public class JavaFile {
 	private final Path javaFilePath;
 	private final CompilationUnit javaFileAST;
 
-	public JavaFile(Path javaFilePath, CompilationUnit javaFileAST) {
+	public JavaFile(Path javaFilePath, CompilationUnit javaFileAST, boolean excludeMainMethod) {
 		this.javaFilePath = javaFilePath;
+		if (excludeMainMethod) {
+			excludeMainMethod(javaFileAST);
+		}
 		this.javaFileAST = javaFileAST;
 	}
 
+	private static void excludeMainMethod(CompilationUnit javaFileAST) {
+		javaFileAST.findAll(MethodDeclaration.class).stream()
+				.filter(method -> method.isStatic() && method.getNameAsString().equals("main")
+						&& method.getParameters().size() == 1 && method.getType().isVoidType()
+						&& method.getParameter(0).getTypeAsString().equals("String[]"))
+				.forEach(Node::remove);
```

Comment:
We modify the AST here, it seems. Are we absolutely sure this has no unintended side effects?

Reply:
Or put differently: what does the removal do? Both according to the documentation and the actual implementation?

Reply:
Removal removes the node from the tree and it's child nodes. This seems to be working as intented in the documentation and the actual implementation.